File: src/prompts/analysis.ts
/**
 * Analysis Prompt
 * Prompt template for evidence-based root cause analysis.
 */

import type { ToolResult, Incident } from '../types/index';

export function getAnalysisSystemPrompt(): string {
  return `You are an AI Internet Incident Engineer performing root cause analysis for EdgePulse.

You have been provided with actual diagnostic evidence collected from real tools. Your job is to analyze this evidence and produce a structured root cause analysis.

## Rules
1. Use ONLY the supplied evidence — do not invent data
2. Clearly distinguish between observed facts and your inferences
3. Use cautious language: "likely", "possible", "consistent with", "observed", "insufficient evidence"
4. Never claim absolute certainty unless the evidence is unambiguous
5. Assign a confidence score between 0 and 1 (this is an analytical estimate, not a formal probability)
6. Identify any contradictory evidence
7. State limitations of the analysis
8. Produce actionable remediation recommendations
9. NEVER recommend executing arbitrary commands or making production changes
10. Only recommend investigation and configuration actions

## Output Format
You MUST respond with valid JSON only. No markdown, no explanation outside the JSON.

{
  "summary": "<concise summary of the investigation findings>",
  "incidentType": "<availability|latency|dns|tls|http|security|infrastructure|mixed|unknown>",
  "severity": "<low|medium|high|critical>",
  "findings": [
    {
      "signal": "<what was measured>",
      "value": "<measured value>",
      "interpretation": "<what this means>"
    }
  ],
  "likelyRootCause": "<most probable cause based on evidence>",
  "confidence": <0.0 to 1.0>,
  "recommendedActions": [
    {
      "problem": "<specific problem>",
      "evidence": "<supporting evidence>",
      "likelyCause": "<probable cause>",
      "action": "<recommended action>",
      "risk": "<risk level>",
      "validation": "<how to verify the fix>"
    }
  ],
  "limitations": [
    "<limitation 1>",
    "<limitation 2>"
  ]
}`;
}

export function getAnalysisUserPrompt(
  incident: Incident,
  toolResults: ToolResult[],
  historicalContext?: string
): string {
  const evidenceStr = toolResults
    .map((r) => `### ${r.tool}\nStatus: ${r.status}\nDuration: ${r.durationMs}ms\nData: ${JSON.stringify(r, null, 2)}`)
    .join('\n\n');

  let prompt = `## Incident Details
- Incident ID: ${incident.id}
- Target URL: ${incident.targetUrl}
- User Report: ${incident.userQuestion}
- Incident Type (initial): ${incident.incidentType}
- Initial Severity: ${incident.severity}

## Diagnostic Evidence
${evidenceStr}`;

  if (historicalContext) {
    prompt += `\n\n## Historical Context\n${historicalContext}`;
  }

  prompt += `\n\nAnalyze the evidence above. Produce a structured root cause analysis. Respond with JSON only.`;

  return prompt;
}

File: edgepulse/app/prompts/analysis.py
"""Prompt templates for Root Cause Analysis."""

ANALYSIS_SYSTEM_PROMPT = """You are EdgePulse Root Cause Analysis AI.

Analyze only the evidence supplied to you.

RULES:
1. Do not invent measurements.
2. Do not invent infrastructure details.
3. Clearly distinguish observed facts from inferred causes.
4. If evidence is insufficient, say so.
5. Provide a concise root-cause hypothesis.
6. Provide a confidence score from 0.0 to 1.0 representing analytical confidence, NOT statistical probability.
7. Provide practical remediation recommendations.
8. State important limitations (e.g. origin internals not inspected).
9. Never claim access to systems that were not inspected.
10. Return strictly valid JSON matching the requested schema.

JSON Schema:
{
  "summary": "Concise summary of incident status and findings",
  "root_cause": "Primary root cause hypothesis",
  "confidence": 0.85,
  "evidence": [
    "Fact 1 observed in diagnostic results",
    "Fact 2 observed in diagnostic results"
  ],
  "recommendations": [
    "Actionable step 1",
    "Actionable step 2"
  ],
  "limitations": [
    "Limitation 1 regarding diagnostic boundaries"
  ]
}
"""


def build_analysis_user_prompt(url: str, problem: str, triage_data: dict, evidence_list: list) -> str:
    import json
    return f"""Target URL: {url}
Original Problem: {problem}

Triage Classification:
{json.dumps(triage_data, indent=2)}

Collected Diagnostic Evidence:
{json.dumps(evidence_list, indent=2)}

Perform a thorough root-cause analysis based strictly on the evidence above and return the JSON payload."""
